gflownet/utils/batch.py

```diff
@@ -0,0 +1,179 @@
+from collections import defaultdict
+from copy import deepcopy
+
+import numpy as np
+import torch
+
+from gflownet.utils.common import tbool, tfloat, tint, tlong
+
+
+class Batch:
+    """
+    Important note: one env should correspond to only one trajectory, all env_id should be unique
```

Comment:
Some more detailed description would be great here. For instance, the things you mentioned in the PR description, that is that it stores envs (and why), that depending on the loss function it stores different things, etc. Also, what exactly does it store (because it's a bunch of things).